### PR TITLE
MM-36547 Renamed 'Mute Channel' and 'Edit Channel Header' menus

### PR DESCRIPTION
**Pull request by lordinkavu**
#### Summary

Renamed 'Mute Channel' and 'Edit Channel Header' menus in Group and Direct Messages to 'Mute Conversation' and 'Edit Conversation Header'.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/17797
JIRA ticket MM-36547
#### Screenshot 
Screenshot 2021-06-28 at 13-59-12 (2) ingavu - sample Mattermost
```release-note
Small text change in direct and group messaging menus. ' Mute channel ' and ' Edit Channel Header ' now reads ' Mute Conversation ' and ' Edit Conversation Header ' .
```

### CODE REVIEW

**Comment by Willyfrog** [comment]
hello @lordinkavu and thanks for submitting the PR! can you check the linter, seems to be throwing an error.

File: components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx [hunk 1]
```diff
@@ -65,13 +64,22 @@ export default class MenuItemToggleMuteChannel extends React.PureComponent<Props
...
+            } else {
+                text = localizeMessage('channel_header.muteConversation', 'Mute Conversation');
+            }
         } else {
-            text = localizeMessage('channel_header.mute', 'Mute Channel');
+            if (isMuted) {
+                text = localizeMessage('channel_header.unmute', 'Unmute Channel');
+            } else {
+                text = localizeMessage('channel_header.mute', 'Mute Channel');
+            }
```
Comment (michelengelen) [inline components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx:82]: We should avoid nesting `if/else` statements as much as possible. I can see that this is not easily done here, but we could at least simplify the assignment by using a ternary operator

```typescript
        if (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL) {
            text = isMuted ? 
                localizeMessage('channel_header.unmuteConversation', 'Unmute Conversation') :
                localizeMessage('channel_header.muteConversation', 'Mute Conversation');
        } else {
            text = isMuted ?
                localizeMessage('channel_header.unmute', 'Unmute Channel') :
                localizeMessage('channel_header.mute', 'Mute Channel');
        }
```

**Review by michelengelen — CHANGES_REQUESTED** [review]
@lordinkavu just a small change request. 

If you could do that and fix the failing tests we should be good to go.

**Comment by michelengelen** [comment]
I nearly forgot: @enahum is there already a similar ticket for the mobile app? Maybe @lordinkavu would be interested in adding this to the mobile repo as well?

File: components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx (hunk 1 above)
Comment (lordinkavu) [reply components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx:82]: Yeah will do that :+1:

**Comment by lordinkavu** [comment]
> I nearly forgot: @enahum is there already a similar ticket for the mobile app? Maybe @lordinkavu would be interested in adding this to the mobile repo as well?

Yes I can try :)

**Review by michelengelen — APPROVED** [review]
LGTM .. thanks @lordinkavu 

And it would be really helpful if you could do the mobile change as well! 👍🏼

**Comment by lordinkavu** [comment]
> LGTM .. thanks @lordinkavu
> 
> And it would be really helpful if you could do the mobile change as well! 👍🏼

Yes. Please assign the issue to me :)

**Comment by enahum** [comment]
> I nearly forgot: @enahum is there already a similar ticket for the mobile app? Maybe @lordinkavu would be interested in adding this to the mobile repo as well?

If it is not the same tix, we should be asking @esethna

**Comment by esethna** [comment]
@lordinkavu Feel free to submit a mobile PR (I'll add mobile to the issue description so it's explicit)

**Comment by lordinkavu** [comment]
> @lordinkavu Feel free to submit a mobile PR (I'll add mobile to the issue description so it's explicit)

**Comment by lordinkavu** [comment]
> @lordinkavu Feel free to submit a mobile PR (I'll add mobile to the issue description so it's explicit)

Okay!

**Review by michelengelen — APPROVED** [review]
LGTM

**Review by Willyfrog — APPROVED** [review]
LGTM!

**Review by jgilliam17 — APPROVED** [review]
Thank you @lordinkavu 
Tested, looks good to merge.
- Verified `Mute Channel`, `Unmute Channel` and `Edit Channel Header` menus in Group and Direct Messages are changed to `Mute Conversation`, `Unmute Conversation` and `Edit Conversation Header`.

**Pull request merged**